src/codelabs/iterables.md

```diff
@@ -0,0 +1,608 @@
+<iframe
+  src="{{site.dartpad-embed}}?id=d56963729339cea951e16209e0a26e4c"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="375"
+  width="100%" >
+</iframe>
+
+In the example, `any()` verifies that at least one element contains the
+character `a`, and `every()` verifies that all elements have a length equal to 
+or greater than 5.
+
+After running the code, try changing the predicate of `any()` so it
+returns false:
+
+<?code-excerpt "iterables/test/iterables_test.dart (any-false)"?>
+{% prettify dart %}
+if (items.any((element) => element.contains('Z'))) {
+  print('At least one element contains "Z"');
+} else {
+  print('No element contains "Z"');
+}
+{% endprettify %}
+
+You can also use `any()` to verify that no element of an `Iterable`
+matches a certain condition.
+
+
+### Exercise: Verify that an Iterable satisfies a condition
+
+The following exercise provides practice using the `any()` and `every()` methods, described
+in the previous example. In this case, there's a list of users with the class `User`
+with the member field `age`.
+
+<?code-excerpt "iterables/lib/user.dart"?>
+{% prettify dart %}
+class User {
+  String name;
+  int age;
+}
+{% endprettify %}
+
+Use `any()` and `every()` to implement two functions:
+
+* Part 1: Implement `anyUserUnder18()`.
+  * Return `true` if at least one user is 17 or younger.
+* Part 2: Implement `everyUserOver13()`.
+  * Return `true` if all users are 14 or older.
+
+<!-- [//]: https://gist.github.com/76dc5c8644652adf4e2032912d974ac3 -->
+
+<iframe
+src="{{site.dartpad-embed}}?id=76dc5c8644652adf4e2032912d974ac3&theme=dark"
+frameborder="no"
+height="325"
+width="100%" >
+</iframe>
+
+{{site.alert.secondary}}
+  **Quick review:**
+  * Although you can use `for-in` loops to check conditions, there are better ways to do that.
+  * The method `any()` enables you to check if any element matches a condition.
+  * The method `every()` enables you to verify if all elements match a condition.
+{{site.alert.end}}
+
+## Filtering
+
+In the previous sections, you saw how using methods like `firstWhere()` or
+`singleWhere()` can help you find an element that matches a certain predicate.
+
+But, what if you want to find all the elements that match a certain condition?
+You can accomplish that using the `where()` method.
+
+<?code-excerpt "iterables/test/iterables_test.dart (where)"?>
+{% prettify dart %}
+Iterable<int> evenNumbers = numbers.where((number) => number.isEven);
+{% endprettify %}
+
+In this example, `numbers` contains an `Iterable` with multiple `int`, and 
+`where()` finds all the numbers that are even.
+
+The output of `where()` is another `Iterable`, and you can use it as such to iterate
+over it or apply other `Iterable` methods. In the next example, the output of `where()`
+is used directly inside the `for-in` loop.
+
+<?code-excerpt "iterables/test/iterables_test.dart (where-for)"?>
+{% prettify dart %}
```

Comment:
Nit: 

Since it's not much to add, should we consider making this example runnable as is? In other words:

```dart
const numbers = [1,2,3,4,5];
Iterable<int> evenNumbers = numbers.where((number) => number.isEven);

main() {
  for (var number in evenNumbers) {
    print('$number is even');
  }
}
```